flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/MLUtils.scala

```diff
@@ -119,4 +123,65 @@ object MLUtils {
 
     stringRepresentation.writeAsText(filePath)
   }
+
+  /** Create a [[ContinuousHistogram]] from the input data
+    *
+    * @param bins Number of bins required
+    * @param data input [[DataSet]] of [[Double]]
+    * @return [[ContinuousHistogram]] over the data
+    */
+  def createContinuousHistogram(data: DataSet[Double], bins: Int): DataSet[ContinuousHistogram] = {
+    val min = data.reduce((x, y) => Math.min(x, y))
+    val max = data.reduce((x, y) => Math.max(x, y))
+
+    val stats = min.mapWithBcVariable(max) {
+      (minimum, maximum) => (minimum - 2 * (maximum - minimum), maximum + 2 * (maximum - minimum))
+    }
+
+    data.mapPartition(new RichMapPartitionFunction[Double, ContinuousHistogram] {
```

Comment:
We could write this a little bit more scalaesque if we introduce a `mapPartitionWithBcVariable` function.